### PR TITLE
Event changes

### PR DESCRIPTION
**Pull request by gauntface**
This changes / adds events:

- statuschange
- requestingpermission
- requestingsubscription

statuschange is essentially fired when the subscription state or permission state changes and has the following data:

    {
        isSubscribed: (results[0] !== null),
        currentSubscription: results[0],
        permissionStatus: results[1].state
      }

This PR also has some subtle edge case fixes and demos of the various approaches.

cc @wibblymat @jeffposnick @addyosmani and @paullewis (these are some of the changes we discussed during devrel summit)

### CODE REVIEW

File: test/browser-tests/push-client/push-client-tests.js [hunk 1]
```diff
@@ -31,17 +31,17 @@ describe('Test PushClient', () => {
   describe('Test PushClient construction', () => {
     it('should be able to create a new push client', function() {
       var pushClient = new window.goog.propel.Client();
-      window.chai.expect(pushClient.endpoint).to.eql(null);
-      window.chai.expect(pushClient.userId).to.equal(null);
-      window.chai.expect(pushClient.workerUrl).to.contain('dist/worker.js');
+      window.chai.expect(pushClient._endpoint).to.equal(null);
```
Comment (wibblymat) [inline test/browser-tests/push-client/push-client-tests.js]: `_endpoint` and `_userId` don't exist any more.

**Comment by wibblymat** [comment]
I keep thinking I've reviewed this, and I haven't. Sorry!

Anyway, now I have!

Code changes LGTM, but the tests need updating to pass after the recent changes.

**Comment by gauntface** [comment]
Thanks for the review @wibblymat.

I've started work on a set of proper tests for propel client in a seperate branch, will hopefully having a PR to show next week.

In short term I've altered the constructor tests to pass for now.
